doc/gc.rst

```diff
@@ -75,6 +75,208 @@ JavaScript's garbage collector is used for the `JavaScript and NodeJS
 The `NimScript <nims.html>`_ target uses the memory management strategy built into
 the Nim compiler.
 
+ARC
+===
+
+ARC, which is short for automatic reference counting, is a memory-management
+strategy based on plain reference counting. Each managed heap cell (that is,
+a location allocated via either `new` or by using the object construction syntax
+where the type is a `ref T`) also stores a reference counter.
+
+When a managed cell is allocated, its counter starts at `1` (because the
+allocation created a single `ref` handle). Creating a `ref` to the cell
+increments the counter, and destroying one decrements it It is incremented when
+a new `ref` to the cell is created, and decremented when a `ref` to it is
+destroyed.
```

Comment:
```suggestion
When a managed cell is allocated, its counter starts at `1` (because the
allocation created a single `ref` handle). Creating a `ref` to the cell
increments the counter, and destroying one decrements it.
```